lib/string-checker.js

```diff
@@ -130,7 +130,12 @@ StringChecker.prototype = {
 
         this._configuredRules.forEach(function(rule) {
             errors.setCurrentRule(rule.getOptionName());
-            rule.check(file, errors);
+
+            try {
+                rule.check(file, errors);
+            } catch (e) {
+                errors.add(e, 1, 0);
```

Comment:
If that is only use-case for passing whole error object, i'd say we shouldn't complicate the `Errors#add`, just `<your error> + error.stack` is enough. 

Warnings would be nice to have for this case and for rule deprecation but it would mean changing reporters API.